Tasks/OperationResult.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Text;
+
+namespace DUCK.Tasks
+{
+	public class OperationResult
+	{
+		public bool IsError { get; set; }
+		public Error Error { get; set; }
```

Comment:
why public set?

Reply:
It's just a result object. You can create it however you wish. There are a bunch of helper functions to do it, but if you would rather construct the object yourself you are not restricted from doing it that way